internal/api/handler/oscal/inventory.go

```diff
@@ -0,0 +1,544 @@
+		})
+	}
+	
+	return nil
+}
+
+func (h *InventoryHandler) applyFilters(items []InventoryItemWithSource, req GetAllInventoryItemsRequest) []InventoryItemWithSource {
+	filtered := items
+
+	// Filter by item type if specified
+	if req.ItemType != "" {
+		var typeFiltered []InventoryItemWithSource
+		for _, item := range filtered {
+			// Check if the item has a property with the name "asset-type" matching the requested type
+			if item.Props != nil {
+				for _, prop := range *item.Props {
+					if prop.Name == "asset-type" && prop.Value == req.ItemType {
+						typeFiltered = append(typeFiltered, item)
+						break
+					}
+				}
+			}
+		}
+		filtered = typeFiltered
+	}
+
+	// Additional filtering by SSP attachment status
+	if req.AttachedToSSP != "" {
+		var attachmentFiltered []InventoryItemWithSource
+		attachToSSP := req.AttachedToSSP == "true"
+		for _, item := range filtered {
+			isAttached := item.SourceType == "ssp"
+			if (attachToSSP && isAttached) || (!attachToSSP && !isAttached) {
+				attachmentFiltered = append(attachmentFiltered, item)
+			}
+		}
+		filtered = attachmentFiltered
+	}
+
+	return filtered
+}
+
+// GetInventoryItem godoc
+//
+//	@Summary		Get Inventory Item by ID
+//	@Description	Retrieves a specific inventory item by its ID
+//	@Tags			Inventory
+//	@Produce		json
+//	@Param			id	path		string	true	"Inventory Item ID"
+//	@Success		200	{object}	InventoryItemWithSource
+//	@Failure		400	{object}	api.Error
+//	@Failure		401	{object}	api.Error
+//	@Failure		404	{object}	api.Error
+//	@Failure		500	{object}	api.Error
+//	@Security		OAuth2Password
+//	@Router			/oscal/inventory/{id} [get]
+func (h *InventoryHandler) GetInventoryItem(ctx echo.Context) error {
+	idParam := ctx.Param("id")
+	id, err := uuid.Parse(idParam)
+	if err != nil {
+		h.sugar.Warnw("Invalid inventory item id", "id", idParam, "error", err)
+		return ctx.JSON(http.StatusBadRequest, api.NewError(err))
+	}
+
+	var item relational.InventoryItem
+	if err := h.db.Preload("ImplementedComponents").First(&item, "id = ?", id).Error; err != nil {
+		if err == gorm.ErrRecordNotFound {
+			return ctx.JSON(http.StatusNotFound, api.NewError(err))
+		}
+		h.sugar.Errorw("Failed to get inventory item", "id", idParam, "error", err)
+		return ctx.JSON(http.StatusInternalServerError, api.NewError(err))
+	}
+
+	oscalItem := item.MarshalOscal()
+	
+	// Determine source
+	source := "Unknown"
+	sourceType := "unknown"
+	if item.SystemImplementationId != uuid.Nil {
+		source = "System Security Plan"
+		sourceType = "ssp"
+	} else {
+		// Check if it's from evidence
+		var count int64
+		h.db.Table("evidence_inventory_items").Where("inventory_item_id = ?", id).Count(&count)
+		if count > 0 {
+			source = "Evidence Collection"
+			sourceType = "evidence"
+		}
+	}
+
+	response := InventoryItemWithSource{
+		InventoryItem: oscalItem,
+		Source:       source,
+		SourceID:     item.ID.String(),
+		SourceType:   sourceType,
+	}
+
+	return ctx.JSON(http.StatusOK, response)
```

Comment:
These should be wrapped in the generic data response and the swagger to be updated

```suggestion
	return ctx.JSON(http.StatusOK, handler.GenericDataResponse[InventoryItemWithSource]{Data: response})
```